src/test/java/teammates/test/cases/ui/pagedata/StudentCommentsPageDataTest.java

```diff
@@ -0,0 +1,339 @@
+        
+        // JSTL data structure comparison: Feedback session rows
+        List<FeedbackSessionRow> actualFeedbackSessionRows = data.getFeedbackSessionRows();
+        assertEquals(1, actualFeedbackSessionRows.size());
+        FeedbackSessionRow expectedFeedbackSessionRow = getFeedbackSessionRow(bundle);
+        FeedbackSessionRow actualFeedbackSessionRow = actualFeedbackSessionRows.get(0);
+        checkFeedbackSessionRowsEqual(expectedFeedbackSessionRow, actualFeedbackSessionRow);
+    }
+
+    private static CommentsForStudentsTable getCommentsForStudentsTable(
+            String giverDetails, String studentEmail, List<CommentAttributes> comments, CourseRoster roster) {
+        List<Comment> commentRows = new ArrayList<Comment>();
+        for (CommentAttributes comment : comments) {
+            String recipientDetails = getRecipientNames(comment.recipients, studentEmail, roster);
+            commentRows.add(new Comment(comment, giverDetails, recipientDetails));
+        }
+        CommentsForStudentsTable commentsForStudentsTable = 
+                new CommentsForStudentsTable(giverDetails, commentRows);
+        return commentsForStudentsTable;
+    }
+    
+    private static FeedbackSessionRow getFeedbackSessionRow(FeedbackSessionResultsBundle bundle) {
+        List<QuestionTable> questionTables = new ArrayList<QuestionTable>();
+        FeedbackSessionAttributes session = bundle.feedbackSession;
+        Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> questionToResponsesMap =
+                bundle.getQuestionResponseMap();
+        for (FeedbackQuestionAttributes question : questionToResponsesMap.keySet()) {
+            List<ResponseRow> responseRows = new ArrayList<ResponseRow>();
+            List<FeedbackResponseAttributes> responses = questionToResponsesMap.get(question);
+            for (FeedbackResponseAttributes response : responses) {
+                List<FeedbackResponseComment> feedbackResponseCommentRows = new ArrayList<FeedbackResponseComment>();
+                List<FeedbackResponseCommentAttributes> responseComments = 
+                        bundle.responseComments.get(response.getId());
+                for (FeedbackResponseCommentAttributes responseComment : responseComments) {
+                    FeedbackResponseComment feedbackResponseCommentRow = 
+                            new FeedbackResponseComment(responseComment, responseComment.giverEmail);
+                    feedbackResponseCommentRows.add(feedbackResponseCommentRow);
+                }
+                String giverName = bundle.getGiverNameForResponse(question, response);
+                String giverTeamName = bundle.getTeamNameForEmail(response.giverEmail);
+                giverName = bundle.appendTeamNameToName(giverName, giverTeamName);
+    
+                String recipientName = 
+                        bundle.getRecipientNameForResponse(question, response);
```

Comment:
I'm actually reluctant to comment on this kind of stuff, but... I don't think this needs line break :stuck_out_tongue: